### ISSUE TITLE
Channels depending on an other channel's value

### ISSUE THREAD

**Issue by fxedel** [mcallegari/qlcplus#989]
Have a look at the American DJ Revo 4 (in 4ch mode): http://adjmedia.s3-website-eu-west-1.amazonaws.com/manuals/revo_4_ir.pdf:
- Channel 1 is Function and can be set to Pattern or Chase
- If Channel 1 is Pattern, Channel 3 is Color (with some capabilities for different colors)
- If Channel 1 is Chase, Channel 3 is Chase Speed (0-100%)

So the behavior of Channel 3 is dependent of Channel 1's value. Currently, QLC+ can't handle this and you have to decide for one of both channel behaviors (Color or Chase Speed):
```xml
 <Channel Name="Function">
  <Group Byte="0">Beam</Group>
  <Capability Min="0" Max="9">Off</Capability>
  <Capability Min="10" Max="99">Pattern</Capability>
  <Capability Min="100" Max="239">Chase</Capability>
  <Capability Min="240" Max="255">Sound active</Capability>
</Channel>
...
 <Channel Name="Colour (Chase Speed)">
  <Group Byte="0">Colour</Group>
  <Capability Min="0" Max="17">R (Slow)</Capability>
  <Capability Min="18" Max="34">G</Capability>
  ...
  <Capability Min="222" Max="238">GBW</Capability>
  <Capability Min="239" Max="255">RGBW (Fast)</Capability>
</Channel>
```
This is not a perfect solution. For example, when I've got buttons for Pattern/Chase and the different Colors and a slider for Chase Speed, I can't control Color and Chase Speed independently from each other, so whenever I switch from Pattern to Chase, I have to adjust the slider and when I switch from Chase to Pattern, I have to select the correct color.

If would be perfect to handle Channel 3 like 2 different channels:
- Color, only active if Channel 1 is set to 0-99
- Chase Speed, only active if Channel 1 is set to 100-255

Then, you'd have separate faders for both channel behaviors, and only one of them is used at the same time.

---

When we added fixtures like this to the [Open Fixture Library](https://github.com/FloEdelmann/open-fixture-library), we came up with the idea of Switching Channels:

```js
"Function": { // Channel 1
  "type": "Effect",
  "defaultValue": 0,
  "capabilities": [
    // ...
    {
      "range": [10, 99],
      "name": "Pattern",
      "switchChannels": {
        "Color / Chase Speed": "Color" // Set Channel 3 to Color
      }
    },
    {
      "range": [100, 239],
      "name": "Chase",
      "switchChannels": {
        "Color / Chase Speed": "Chase Speed"  // Set Channel 3 to Chase Speed
      }
    }
    // ...
  ]
}
// ...
"modes": [
  {
    "name": "4 channel",
    "shortName": "4ch",
    "channels": [
      "Function",
      "Pattern / Chase",
      "Color / Chase Speed", // this is the alias for the channel that can either be Color or Chase Speed
      "Strobe"
    ]
  }
]
```
See here: [Revo 4 IR](http://open-fixture-library.herokuapp.com/american-dj/revo-4-ir)

Maybe it'll be worth implementing something similar in QLC+?

**Comment by janosvitok** [comment]
FYI1: I've started collection special fixtures here: https://github.com/mcallegari/qlcplus/wiki/Special-Fixtures
I've added this case there.

FYI2: There was a Chinese fixture that had different number of DMX channels depending on first channel value. basically, the first channel was used to switch personalities. I will try to find it out.

**Comment by mcallegari** [comment]
This can be closed now. Addressed with #1026
